src/intree.js

```diff
@@ -103,72 +104,101 @@ module.exports.setup = async function({cfg, schemaset}) {
     // Perform parameter substitutions. This happens after verification
     // because templating may change with schema version, and parameter
     // functions are used as default values for some fields.
-    config = jparam(config, _.merge(payload.details, {
-      $fromNow: (text) => tc.fromNowJSON(text),
-      timestamp: Math.floor(new Date()),
-      organization: payload.organization,
-      repository: payload.repository,
-      'taskcluster.docker.provisionerId': cfg.intree.provisionerId,
-      'taskcluster.docker.workerType': cfg.intree.workerType,
-    }));
+    if (version === 0) { // TODO: version 0 stuff to make a separate module
+      config = jparam(config, _.merge(payload.details, {
+        $fromNow: (text) => tc.fromNowJSON(text),
+        timestamp: Math.floor(new Date()),
+        organization: payload.organization,
+        repository: payload.repository,
+        'taskcluster.docker.provisionerId': cfg.intree.provisionerId,
+        'taskcluster.docker.workerType': cfg.intree.workerType,
+      }));
+    } else {
+      if (!branchTest.test(payload.branch || '')) {
+        throw new Error('Cannot have unicode in branch names!');
+      }
+      if (!branchTest.test(payload.branch || '')) {
+        throw new Error('Cannot have unicode in branch names!');
+      }
+
+      config = jsone(config, {
+        tasks_for: payload.tasks_for,
+        event: payload.body,
+        branch: payload.branch,
```

Comment:
I'd like to not (yet?) have a branch in the context.  Removing this will just mean removing `branch` from some of the `$if` conditions in the fixtures.  If some of those fixtures become redundant, it's OK to drop them -- v1 is simpler than v0, so it makes sense that it requires fewer tests.